pkg/cmd/adm/register_member.go

```diff
@@ -120,66 +114,207 @@ func registerMemberCluster(ctx *extendedCommandContext, newCommand client.Comman
+	//
+	// create/update toolchaincluster on the targetCluster
+	ctx.Printlnf("creating ToolchainCluster representation of %s in %s:", sourceClusterType, targetClusterDetails.toolchainClusterName)
+	toolchainClusterCR := &toolchainv1alpha1.ToolchainCluster{ObjectMeta: metav1.ObjectMeta{Name: sourceClusterDetails.toolchainClusterName, Namespace: targetClusterDetails.namespace}}
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), targetClusterDetails.client, toolchainClusterCR, func() error {
+
+		// update the tc label
+		labels := toolchainClusterCR.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		// TODO drop this "namespace" label as soon as ToolchainCluster controller supports loading data from kubeconfig
+		labels["namespace"] = sourceClusterDetails.namespace
+		toolchainClusterCR.Labels = labels
+		toolchainClusterCR.Spec.APIEndpoint = sourceClusterDetails.apiEndpoint
+		toolchainClusterCR.Spec.SecretRef.Name = secretName
+		if insecureSkipTLSVerify {
+			toolchainClusterCR.Spec.DisabledTLSValidations = []toolchainv1alpha1.TLSValidation{toolchainv1alpha1.TLSAll}
+		}
+
+		return nil
+	})
 
-	script, err := downloadScript(term)
 	if err != nil {
 		return err
 	}
-	args := []string{script.Name(), "--type", joiningClusterType.String(), "--host-kubeconfig", hostKubeconfig, "--host-ns", hostNs, "--member-kubeconfig", memberKubeconfig, "--member-ns", memberNs}
-	if len(nameSuffix) > 0 {
-		args = append(args, "--multi-member", nameSuffix)
+	ctx.Println("Toolchaincluster successfully reconciled")
+	toolchainClusterKey := runtimeclient.ObjectKey{
+		Name:      sourceClusterDetails.toolchainClusterName,
+		Namespace: targetClusterDetails.namespace,
 	}
-	if useLetsEncrypt {
-		args = append(args, "--lets-encrypt")
+	if err := waitUntilToolchainClusterReady(ctx.CommandContext, targetClusterDetails.client, toolchainClusterKey, v.args.waitForReadyTimeout); err != nil {
+		ctx.Printlnf("The ToolchainCluster resource representing the %s in the %s cluster has not become ready.", sourceClusterType, sourceClusterType.TheOtherType())
+		ctx.Printlnf("Please check the %s ToolchainCluster resource in the %s %s cluster.", toolchainClusterKey, targetClusterDetails.apiEndpoint, sourceClusterType.TheOtherType())
+		return err
 	}
-	term.Printlnf("Command to be called: bash %s\n", strings.Join(args, " "))
-	bash := newCommand("bash", args...)
-	bash.Stdout = os.Stdout
-	bash.Stderr = os.Stderr
-	return bash.Run()
+	// -- end temporary logic
+
+	return nil
 }
 
-func downloadScript(term ioutils.Terminal) (*os.File, error) {
-	resp, err := http.Get(AddClusterScriptURL)
-	if err != nil {
-		return nil, errs.Wrapf(err, "unable to get add-script.sh")
+func newRestClient(kubeConfigPath string) (*rest.RESTClient, error) {
+	restClientConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	restClientConfig.ContentConfig = rest.ContentConfig{
```

Comment:
Where is this coming from and why do we need this? I would appreciate a comment with an explanation for having to override the configuration obtained from the kubeconfig file.

Reply:
yeah these are apparently required when using the rest client, I see we had to set those in other commands as well. See : https://github.com/kubesaw/ksctl/blob/7f0bcfb34a22559f64decad0be78446154f061a0/pkg/cmd/adm/must_gather_namespace.go#L43-L47

Done in https://github.com/kubesaw/ksctl/pull/53/commits/380c3b6cb4cb13925bde54bbeb74717bd7f7479f